Contest_API.md

```diff
@@ -694,8 +694,8 @@ Properties of contest objects:
 | countdown\_pause\_time       | RELTIME ?       | The amount of seconds left when countdown to contest start is paused. At no time may both `start_time` and `countdown_pause_time` be non-`null`.
 | duration                     | RELTIME         | Length of the contest.
 | scoreboard\_freeze\_duration | RELTIME ?       | How long the scoreboard is frozen before the end of the contest. Defaults to `0:00:00`.
-| scoreboard\_type             | string ?        | What type of scoreboard is used for the contest. Must be either `pass-fail` or `score`. Defaults to `pass-fail`.
-| penalty\_time                | integer ?       | Penalty time for a wrong submission, in minutes. Only relevant if scoreboard\_type is `pass-fail`. Defaults to `0`.
+| scoreboard\_type             | string          | What type of scoreboard is used for the contest. Must be either `pass-fail` or `score`. Defaults to `pass-fail`.
+| penalty\_time                | integer         | Penalty time for a wrong submission, in minutes. Only relevant if scoreboard\_type is `pass-fail`. Defaults to `0`.
```

Comment:
I think both of these should simply be present if specified in `/access`: this is fundamental contest configuration, and there's only a single instance of each of these properties since they are in the contest object. So it should be whether this data is available and the server willing and able to provide it. And if the server doesn't specify these in `/access`, then there's still the default values that a client can rely on.

Reply:
> I don't agree with the "if it's not in `/access` id doesn't exists and if it doesn't exists it can't have a default".
> 
> We agree that you should be allowed to not provide `scoreboard_type`, right? And that that means that scoreboard is `pass-fail`? This is based on the non-existing property having a default.
>
> Furthermore, any future extension surely currently does not exist. And you would definitely want to provide a default for that.

Not having penalty_time in /access doesn't mean that there is a penalty time and it is 0, there just _isn't_ a penalty time. I totally agree that we need to define the _behaviour_ when a property doesn't exist, it's just confusing to me to specify a default for a property that we've said can't have one. Using language like "If it doesn't exist, ..." would make more sense to me.